Foundation/NSURLResponse.swift

```diff
@@ -154,28 +166,192 @@ public class NSHTTPURLResponse : NSURLResponse {
         @abstract Returns the HTTP status code of the receiver. 
         @result The HTTP status code of the receiver. 
     */
-    public var statusCode: Int { NSUnimplemented() }
+    public let statusCode: Int
     
-    /*! 
-        @method allHeaderFields
-        @abstract Returns a dictionary containing all the HTTP header fields
-        of the receiver.
-        @discussion By examining this header dictionary, clients can see
-        the "raw" header information which was reported to the protocol
-        implementation by the HTTP server. This may be of use to
-        sophisticated or special-purpose HTTP clients.
-        @result A dictionary containing all the HTTP header fields of the
-        receiver.
-    */
-    public var allHeaderFields: [NSObject : AnyObject] { NSUnimplemented() }
+    /// Returns a dictionary containing all the HTTP header fields
+    /// of the receiver.
+    ///
+    /// By examining this header dictionary, clients can see
+    /// the "raw" header information which was reported to the protocol
+    /// implementation by the HTTP server. This may be of use to
+    /// sophisticated or special-purpose HTTP clients.
+    ///
+    /// - Returns: A dictionary containing all the HTTP header fields of the
+    /// receiver.
+    ///
+    /// - Important: This is an *experimental* change from the
+    /// `[NSObject: AnyObject]` type that Darwin Foundation uses.
+    public let allHeaderFields: [String: String]
```

Comment:
This doesn't match the behavior of the real Foundation. Specifically, `allHeaderFields` uses a custom `NSDictionary` subclass that's case-insensitive when looking up keys. That case-insensitive behavior is pretty important behavior when dealing with HTTP headers.

It also canonicalizes certain headers, e.g. `Content-Length`, although it doesn't document which headers (besides Content-Length), and the canonicalization really only affects the case where you're displaying the header name so it's not really particularly important.

Given that Swift dictionaries are a value type and not a class, the only way we're going to get the same behavior while still using a dictionary type is to rely on bridging support and using an `NSDictionary` subclass, like the real Foundation does.